tools/terraform/fleetdm_client/fleetdm_client.go

```diff
@@ -0,0 +1,276 @@
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 300 {
+		b, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("failed to create team %s: %s %s", name, resp.Status, string(b))
+	}
+
+	var newTeam TeamGetResponse
+	err = json.NewDecoder(resp.Body).Decode(&newTeam)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return &newTeam, nil
+}
+
+// GetTeam returns the team with the provided ID.
+func (c *FleetDMClient) GetTeam(id int64) (*TeamGetResponse, error) {
+	url := teamPrefix + "/" + strconv.FormatInt(id, 10)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create GET request for %s: %w",
+			url, err)
+	}
+	resp, err := c.do(req, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to GET %s: %w", url, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		b, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("failed to get team %d: %s %s", id, resp.Status, string(b))
+	}
+
+	var team TeamGetResponse
+	err = json.NewDecoder(resp.Body).Decode(&team)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+	return &team, nil
+}
+
+// UpdateTeam updates the team with the provided ID with the provided name and description.
+func (c *FleetDMClient) UpdateTeam(id int64, name, description *string) (*TeamGetResponse, error) {
+	if name == nil && description == nil {
+		return nil, fmt.Errorf("both name and description are nil")
+	}
+
+	url := teamPrefix + "/" + strconv.FormatInt(id, 10)
+	var teamUpdate TeamCreate
+	if name != nil {
+		teamUpdate.Name = *name
+	}
+	if description != nil {
+		teamUpdate.Description = *description
+	}
+	updateJson, err := json.Marshal(teamUpdate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update team body request: %w", err)
+	}
+	req, err := http.NewRequest(http.MethodPatch, url, bytes.NewReader(updateJson))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create PATCH request for %s body %s: %w",
+			url, updateJson, err)
+	}
+	resp, err := c.do(req, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to PATCH %s body %s: %w",
+			url, updateJson, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 300 {
+		b, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("failed PATCH %s body %s: %s %s",
+			url, updateJson, resp.Status, string(b))
+	}
+
+	var newTeam TeamGetResponse
+	err = json.NewDecoder(resp.Body).Decode(&newTeam)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return &newTeam, nil
+}
+
+// UpdateAgentOptions pretends that the agent options are a string, when it's really actually json.
+// Strangely the body that comes back is a team, not just the agent options.
+func (c *FleetDMClient) UpdateAgentOptions(id int64, ao string) (*TeamGetResponse, error) {
+
+	// First verify it's actually json.
+	var aoJson interface{}
+	err := json.Unmarshal([]byte(ao), &aoJson)
```

Comment:
You can use `json.Valid` to that effect: https://pkg.go.dev/encoding/json#Valid

Reply:
Oh, didn't know that one.